src/harness/reference_models/pre_iap_filtering/fss_purge.py

```diff
@@ -0,0 +1,301 @@
+  fss_longitude = fss_record['record']['deploymentParam'][0]['installationParam']['longitude']
+  fss_height = fss_record['record']['deploymentParam'][0]['installationParam']['height']
+  fss_height_type = fss_record['record']['deploymentParam'][0]['installationParam']['heightType']
+  fss_azimuth = fss_record['record']['deploymentParam'][0]['installationParam']['antennaAzimuth']
+  fss_pointing_elevation = - fss_record['record']['deploymentParam'][0]['installationParam']['antennaDowntilt']
+  fss_max_gain_dbi = fss_record['record']['deploymentParam'][0]['installationParam']['antennaGain']
+
+  # Convert the height to AGL if it was AMSL type.
+  if fss_height_type == 'AMSL':
+    fss_height_agl = convertAmslToAgl(fss_latitude, fss_longitude, fss_height)
+  elif fss_height_type == 'AGL':
+    fss_height_agl = fss_height
+  else:
+    raise Exception('Invalid FSS height type.')
+
+  # Populate FssProtectionPoint tuple.
+  fss_entity = interference.FssProtectionPoint(latitude=fss_latitude,
+                                               longitude=fss_longitude,
+                                               height_agl=fss_height_agl,
+                                               max_gain_dbi=fss_max_gain_dbi,
+                                               pointing_azimuth=fss_azimuth,
+                                               pointing_elevation=fss_pointing_elevation)
+  return fss_entity
+
+def generatePurgeListForFssPoint(cbsds, fss_entity):
+  """ Implementation of algorithm described in R2-SGN-29.
+
+  The list of grants to be purged based on the interference caused by their
+  out of band emission are identified and returned.
+  Args:
+    cbsds: List of CbsdData objects that are in the neighborhood to an FSS.
+    fss_entity: The FssProtectionPoint tuple of the FSS for which the purge is to be performed.
+  Returns:
+    List of multiple key-value pairs with keys "grantData", "cbsd" and values as grants to be
+    purged and corresponding cbsd data object respectively.
+  """
+  # List of grants to be purged for the fss_entity.
+  final_purge_list = []
+
+  # List of dictionaries containing the grant_info based on the highest grant from FSS OOBE calculation
+  grants_cbsds_info_for_oobe_calculation = getGrantInfoListForFssOobeCalculation(cbsds)
+
+  # Calculate the threshold value in dBm for the OOBE interference
+  oobe_threshold_value = -129 + 10 * (np.log10(REF_BW/interference.ONE_MHZ) - FSS_OOBE_MARGIN)
+
+  while True:
+
+    # Calculate the OOBE interference value for each grant.
+    calculateOobeInterference(grants_cbsds_info_for_oobe_calculation, fss_entity)
+
+    # Sort the grants based on the calculated interference value.
+    sorted_grants_cbsds_info = sorted(grants_cbsds_info_for_oobe_calculation, key=lambda grant: grant['oobe_interference'])
+    # Find the largest index such that aggregation of the interferences do not cross
```

Comment:
According to the code below, in case aggregation is **equal** (or higher) to the treshold, the grants will be purged.
Therefore ,this line of comment is a bit misleading. 

Proposals (either is ok)

1) correct the comment in line 103 to :
# Find the largest index such that aggregation of the interferences **is not equal or higher** than the OOBE threshold.

2) Or, correct the code so that in case aggregation is **equal** to the treshold, the grants are **not** purged.

Note: in DPA reference model, option 2) is followed